spec/models/agents/post_agent_spec.rb

```diff
@@ -311,8 +311,21 @@
   end
 
   describe "#working?" do
+    it "checks if there was an error" do
+      @checker.error("error")
+      expect(@checker.logs.count).to eq(1)
+      expect(@checker.reload).not_to be_working
+    end
+
+    it "checks if no event has been received" do
+      expect(@checker.logs.count).to eq(0)
+      expect(@checker.last_receive_at).to be_nil
+      expect(@checker.reload).not_to be_working
+    end
+
     it "checks if events have been received within expected receive period" do
-      expect(@checker).not_to be_working
```

Comment:
I think with the latest you made to the `working?` method this test can stay the same as it was before.